plugins/plugin-svelte/plugin.js

```diff
@@ -48,7 +50,7 @@ module.exports = function plugin(snowpackConfig, pluginOptions = {}) {
 
       const {js, css} = svelte.compile(codeToCompile, {
         ...svelteOptions,
-        generate: isSSR ? 'ssr' : 'dom',
+        generate: svelteOptions.generate || (isSSR ? 'ssr' : 'dom'), // let user-defined option take precedence
```

Comment:
The main bugfix here: if the user has provided this setting, prefer that over Snowpack (remember that users can use `snowpack.config.js` to dynamically set this, and they probably have a better understanding on when this should be on anyway)

Reply:
tbh I'd rather just move the generate line above ..svelteOptions:

```
// Default the generate mode, based on isSSR
generate: isSSR ? 'ssr' : 'dom',
// Apply svelte options
...svelteOptions,
// Apply build options
outputFilename: filePath,
filename: filePath,
```